src/protocol.rs

```diff
@@ -339,9 +341,34 @@ impl ParticipantIdentifier {
     }
 }
 
-impl ProofContext for Vec<ParticipantIdentifier> {
+pub struct ProtocolContext {
```

Comment:
todo: this won't necessarily be the "complete" protocol context because auxinfo and presign will both have inputs from the previous protocols. Can you name this something like `FixedContext` or `ParameterContext` (not loving either of those) to indicate that this is the shared/fixed context determined by known parameters but is not necessarily the complete protocol context?

Reply:
Maybe we should name it`SharedContext` and add a comment?

Reply:
Yes, good call on documentation!